go/vt/vtctl/workflow/framework_test.go

```diff
@@ -198,30 +198,38 @@ func (env *testEnv) addTablet(t *testing.T, ctx context.Context, id int, keyspac
 	return tablet
 }
 
-// addTableRoutingRules adds routing rules from the test env's source keyspace to
-// its target keyspace for the given tablet types and tables.
-func (env *testEnv) addTableRoutingRules(t *testing.T, ctx context.Context, tabletTypes []topodatapb.TabletType, tables []string) {
-	ks := env.targetKeyspace.KeyspaceName
-	rules := make(map[string][]string, len(tables)*(len(tabletTypes)*3))
+func (env *testEnv) saveRoutingRules(t *testing.T, rules map[string][]string) {
+	err := topotools.SaveRoutingRules(context.Background(), env.ts, rules)
+	require.NoError(t, err)
+	err = env.ts.RebuildSrvVSchema(context.Background(), nil)
+	require.NoError(t, err)
+}
+
+func (env *testEnv) updateTableRoutingRules(t *testing.T, ctx context.Context,
+	tabletTypes []topodatapb.TabletType, tables []string, sourceKeyspace, targetKeyspace, toKeyspace string) {
+
+	if len(tabletTypes) == 0 {
+		tabletTypes = []topodatapb.TabletType{topodatapb.TabletType_PRIMARY, topodatapb.TabletType_REPLICA, topodatapb.TabletType_RDONLY}
```

Comment:
IMO nicer to make this a variable called defautTabletTypes since I think that's what we're doing here and individual tests can know this and work with it as well -- better understanding the behavior.